amundsen_application/static/js/components/common/ResourceListItem/TableListItem/tests/index.spec.tsx

```diff
@@ -40,62 +87,82 @@ describe('TableListItem', () => {
       expect(wrapper.find(Link).exists()).toBeTruthy();
     });
 
-    it('renders table name', () => {
-      expect(wrapper.find('.resource-name').children().at(0).text()).toEqual('tableSchema.tableName');
-    });
+    describe('renders resource-info section', () => {
+      let resourceInfo;
+      beforeAll(() => {
+        resourceInfo = wrapper.find('.resource-info');
+      });
 
-    it('renders a bookmark icon', () => {
-      expect(wrapper.find(BookmarkIcon).exists()).toBe(true);
-    });
+      it('renders start correct icon', () => {
+        const startIcon = resourceInfo.find('img');
+        expect(startIcon.exists()).toBe(true);
+        expect(startIcon.props().className).toEqual(wrapper.instance().generateStartIconClass(props.table.database));
+      });
 
-    it('renders table description', () => {
-      expect(wrapper.find('.content').children().at(0).children().at(1).text()).toEqual('I am the description');
-    });
+      it('renders table name', () => {
+        expect(resourceInfo.find('.resource-name').children().at(0).text()).toEqual('tableSchema.tableName');
+      });
+
+      it('renders a bookmark icon in the resource name', () => {
+        expect(resourceInfo.find('.resource-name').find(BookmarkIcon).exists()).toBe(true);
+      });
 
-    it('renders resource type', () => {
-      expect(wrapper.find('.content').children().at(1).text()).toEqual(props.table.database);
+      it('renders table description', () => {
+        expect(resourceInfo.children().at(1).children().at(1).text()).toEqual('I am the description');
```

Comment:
nit: Whenever possible I would prefer to add a class rather than using `children().at(X)`

Reply:
I'll look into this when I do some component cleanup. Ultimately I want to break up the list item into even smaller components. I think a good rule to follow might be to break things down until tests for any component only need to use one level of `children().at(X)` to verify that it's immediate children exist as intended.

I believe that's the best long term solution because the idea of using classes is also leveraging the fact that we know there is only one element with that class. This isn't always guaranteed to be the case. I also consider that this requires every element we are checking to have an explicit class, and I wouldn't want to have to start adding empty classes just to test things.